packages/mds-types/index.ts

```diff
@@ -29,6 +29,8 @@ export type VEHICLE_TYPE = keyof typeof VEHICLE_TYPES
 export const RULE_TYPES = Enum('count', 'speed', 'time')
 export type RULE_TYPE = keyof typeof RULE_TYPES
 
+export type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
```

Comment:
If you move the uuid generation to the API ... this oddly named type won't be necessary I think.

Reply:
done.

Reply:
It's such a snazzy type, though!

Reply:
I understand what it does, but it's name, `PartialBy` is non-intuitive. It feels like `Optional<T, K extends keyof T>` or `WithOptionalProperties<T, K extends keyof T>` would be more accurate. Nonetheless, we don't use it :-) Nice to see whoever did this leveraging mapped types though!